rllib/examples/learners/separate_vf_lr_and_optimizer.py

```diff
@@ -0,0 +1,137 @@
+
+
+Results to expect
+-----------------
+You should expect to observe decent learning behavior from your console output:
+
+With --lr-vf=0.0005 and --lr-policy=0.001
++-----------------------------+------------+-----------------+--------+
+| Trial name                  | status     | loc             |   iter |
+|                             |            |                 |        |
+|-----------------------------+------------+-----------------+--------+
+| PPO_CartPole-v1_7b404_00000 | TERMINATED | 127.0.0.1:16845 |     19 |
++-----------------------------+------------+-----------------+--------+
++------------------+------------------------+---------------------+
+|   total time (s) | num_env_steps_sampled_ | episode_return_mean |
+|                  |              _lifetime |                     |
+|------------------+------------------------+---------------------+
+|          19.4179 |                  76000 |              459.94 |
++------------------+------------------------+---------------------+
+"""
+
+from ray.rllib.algorithms.ppo import PPOConfig
+from ray.rllib.examples.learners.classes.separate_vf_lr_and_optimizer_learner import (
+    PPOTorchLearnerWithSeparateVfOptimizer,
+)
+from ray.rllib.utils.framework import try_import_torch
+from ray.rllib.utils.test_utils import (
+    add_rllib_example_script_args,
+    run_rllib_example_script_experiment,
+)
+
+torch, _ = try_import_torch()
+
+
+parser = add_rllib_example_script_args(default_reward=450.0)
+parser.set_defaults(enable_new_api_stack=True)
+parser.add_argument(
+    "--lr-vf",
+    type=float,
+    default=0.0005,
+    help="The learning rate used in the value function optimizer.",
+)
+parser.add_argument(
+    "--lr-policy",
+    type=float,
+    default=0.001,
+    help="The learning rate used in the policy optimizer.",
+)
+
+
+if __name__ == "__main__":
+    args = parser.parse_args()
+
+    assert (
+        args.enable_new_api_stack
+    ), "Must set --enable-new-api-stack when running this script!"
+    assert args.algo == "PPO", "Must set --algo=PPO when running this script!"
+
+    base_config = (
+        PPOConfig()
+        .environment("CartPole-v1")
+        .training(
+            # This is the most important setting in this script: We point our PPO
+            # algorithm to use the custom Learner (instead of the default
+            # PPOTorchLearner).
+            learner_class=PPOTorchLearnerWithSeparateVfOptimizer,
+            # We use this simple method here to inject a new setting that our
+            # custom Learner class uses in its `configure_optimizers_for_module`
+            # method. This is convenient and avoids having to subclass `PPOConfig` only
+            # to add a few new settings to it. Within our Learner, we can access this
+            # new setting through:
+            # `self.config.learner_config_dict['lr_vf']`
+            learner_config_dict={"lr_vf": args.lr_vf},
```

Comment:
As mentioned before. This is not my taste b/c it increases the number attributes in the `AlgorithmConfig` and makes it less readable than simply overriding it.

Reply:
I get your point. But I do think this is so much simpler. Users often don't care about the intricate design choices of a complex library like RLlib. So they'll roll their eyes if they have to create an extra subclass of AlgorithmConfig (which is a scary thing to override), just to feed a single setting into their learners.

Instead, they just want things to work as easy as possible.
Note that often, users have a) a very limited set of custom options they need to send in and b) know very well, which are their own settings and which are RLlib settings so there is no real danger of confusing these two.

I make the argument here to a) "make simple things simple" and b) "difficult things possible".

If it helps, I can add comments that explain better how you would do it with a custom AlgorithmConfig override as an alternative.

Also keep in mind that the above is analogous to a custom RLModule, for which you also don't have to subclass any config class of ours. You just send in your custom model config dict.

Reply:
Hello @sven1977 @simonsays1980, I'd like to contribute to this discussion as an RLlib user, which might inform your design choices for future RLlib updates. I concur that adding more attributes to `AlgorithmConfig` reduces readability and complicates class modifications for specific use cases. Two years ago, my focus with RLlib usage was on a straightforward interface, without concern for its complexities as @sven1977 has highlighted. Over time, the need for customization became apparent. My point is the necessity for a balance between ease of use and the ability to customize. In essence, I advocate for a) keeping simple tasks uncomplicated, and b) making complex tasks achievable 

> @sven1977 
a) "make simple things simple" and b) "difficult things possible". 

Thank you.